### ISSUE TITLE
make crossorigin prop case insensitive

### ISSUE THREAD

**Issue by codebycarlos** [mbrevda/react-image#967]
First of all thank you so much for this package it's pretty awesome!

If we want to make the api fully compatible with the regular image element we'll need to make the component recognise when users pass in the crossOrigin prop as opposed to just crossorigin. Otherwise, this can lead to confusing behaviour when one sets the crossOrigin prop but it is then ignored (happened to me, probably pretty rare as most people don't use this attribute). I fixed this by passing in the defined crossorigin prop, but either way it should accept both ideally.

**Comment by mbrevda** [comment]
Unclear on the issue here: is the `crossorigin` prop being set on the `<img>` element? Is this purely a lint issue?

**Comment by mbrevda** [comment]
Any props not used by the `Img` component will be passed through to the underlying `<img>` element as HTML attributes, as can be seen [here](https://github.com/mbrevda/react-image/blob/5b6f430847e872eacd8470d7f11599a26f496aeb/src/Img.tsx#L36) and [here](https://github.com/mbrevda/react-image/blob/5b6f430847e872eacd8470d7f11599a26f496aeb/src/Img.tsx#L49). I'm not sure why we would have to special case this attribute vs any other valid HTML attribute?

**Comment by codebycarlos** [comment]
> Any props not used by the `Img` component will be passed through to the underlying `<img>` element as HTML attributes, as can be seen [here](https://github.com/mbrevda/react-image/blob/5b6f430847e872eacd8470d7f11599a26f496aeb/src/Img.tsx#L36) and [here](https://github.com/mbrevda/react-image/blob/5b6f430847e872eacd8470d7f11599a26f496aeb/src/Img.tsx#L49). I'm not sure why we would have to special case this attribute vs any other valid HTML attribute?

That is correct, however, in the case of crossorigin and requesting of the image, this is first handled by [imagePromiseFactory.jsx](https://github.com/mbrevda/react-image/blob/5b6f430847e872eacd8470d7f11599a26f496aeb/src/imagePromiseFactory.ts) via the useImage hook [here](https://github.com/mbrevda/react-image/blob/5b6f430847e872eacd8470d7f11599a26f496aeb/src/Img.tsx#L40). 

By the time the image src is passed to the ```<img />``` element you [mention](https://github.com/mbrevda/react-image/blob/5b6f430847e872eacd8470d7f11599a26f496aeb/src/Img.tsx#L49), crossorigin doesn't make a difference as the image was already requested according to the crossorigin passed (i.e. with Origin header or not depending on its value). 

Basically, the way I noticed this was even an issue when using 'react-image' was this: my team had a situation where we needed to set an image's crossorigin to 'anonymous' (otherwise using it within a canvas would [fail](https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image)). Initially I did something like:

```
   <Img crossOrigin="anonymous"/>
```

Yet the image was still failing, which led me to discover ```<Img />``` from 'react-image' was not picking up crossOrigin, and only crossorigin (unlike a native ```<img />``` element. Upon switching it to 

```
   <Img crossorigin="anonymous" />
```

everything started working again 🎉

**Comment by mbrevda** [comment]
> So the main issue is when you pass the crossOrigin ... it gets ignored

Not really following why this is an issue - the componenet seems to be working as intended?

> This behaviour is different to the native <img /> element which accepts crossOrigin

Oh - you're suggesting that the prop be named in a manner consistent with the native element?
